app/views/helpers/TimelineEvents.scala

```diff
@@ -111,9 +117,27 @@ class TimelineEvents @Inject() (
       }
     }.sorted
 
-  sealed abstract class AmendmentEventIfLatest { val actionId: String }
-  case class AmendmentFailed(actionId: String) extends AmendmentEventIfLatest
-  case class AmendmentRejected(actionId: String) extends AmendmentEventIfLatest
+    if (declarationAmendmentsConfig.isEnabled) {
+      val hasExternalAmendments = allEvents.exists(_.requestType == ExternalAmendmentRequest)
+      if (hasExternalAmendments) allEvents.filterNot { event =>
+        event.requestType == SubmissionRequest && event.notificationSummary.enhancedStatus == AMENDED
+      }
+      else allEvents
```

Comment:
Is this check necessary? Why not just always filter the events (if there are no ExternalAmendmentRequests then nothing will be filtered out)

Reply:
I do not understand. What check do you want to remove?